File: game/app/src/main/java/com/bromobile/game/MainActivity.java
package com.bromobile.game;

import android.app.Activity;
import android.os.Build;
import android.os.Bundle;
import android.view.View;
import android.view.Window;
import android.view.WindowManager;

/**
 * Single-activity host. Everything else lives inside {@link GameView}, which owns
 * the render thread, the game state machine and all input handling.
 */
public class MainActivity extends Activity {

    private GameView view;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        requestWindowFeature(Window.FEATURE_NO_TITLE);
        getWindow().addFlags(WindowManager.LayoutParams.FLAG_KEEP_SCREEN_ON);
        if (Build.VERSION.SDK_INT >= 28) {
            getWindow().getAttributes().layoutInDisplayCutoutMode =
                    WindowManager.LayoutParams.LAYOUT_IN_DISPLAY_CUTOUT_MODE_SHORT_EDGES;
        }
        view = new GameView(this);
        setContentView(view);
        immersive();
    }

    private void immersive() {
        View d = getWindow().getDecorView();
        d.setSystemUiVisibility(
                View.SYSTEM_UI_FLAG_LAYOUT_STABLE
                        | View.SYSTEM_UI_FLAG_LAYOUT_HIDE_NAVIGATION
                        | View.SYSTEM_UI_FLAG_LAYOUT_FULLSCREEN
                        | View.SYSTEM_UI_FLAG_HIDE_NAVIGATION
                        | View.SYSTEM_UI_FLAG_FULLSCREEN
                        | View.SYSTEM_UI_FLAG_IMMERSIVE_STICKY);
    }

    @Override
    public void onWindowFocusChanged(boolean hasFocus) {
        super.onWindowFocusChanged(hasFocus);
        if (hasFocus) immersive();
    }

    @Override
    protected void onResume() {
        super.onResume();
        if (view != null) view.onResumeGame();
        immersive();
    }

    @Override
    protected void onPause() {
        super.onPause();
        if (view != null) view.onPauseGame();
    }

    @Override
    public void onBackPressed() {
        if (view != null && view.handleBack()) return;
        super.onBackPressed();
    }

    /** Called by the game when the player picks "EXIT" in a menu. */
    public void quitGame() {
        finish();
    }
}

File: app/src/main/java/com/discipline/hoeren/MainActivity.kt
package com.discipline.hoeren

import android.os.Bundle
import androidx.activity.ComponentActivity
import androidx.activity.compose.setContent
import com.discipline.hoeren.ui.AppRoot
import com.discipline.hoeren.ui.AppState
import com.discipline.hoeren.ui.theme.DisziplinTheme

class MainActivity : ComponentActivity() {

    private lateinit var state: AppState

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        state = AppState(this)
        setContent {
            DisziplinTheme {
                AppRoot(state)
            }
        }
    }

    /** Возврат в приложение — перечитываем дату и время: день мог смениться. */
    override fun onResume() {
        super.onResume()
        state.refreshNow()
    }
}
